python/sample.py

```diff
@@ -19,8 +20,86 @@
 # limitations under the License.
 import json
 import argparse
+import urllib.request
+import urllib.parse
+
+def get_address(arg_ip_address, use_registry, headers):
+
+    '''
+
+    Gets the address of the Scheduled Events endpoint.
+
+    '''
+
+    ip_address = None
+
+    address = None
+
+    if arg_ip_address:
+
+        # use IP address provided in parameter.
+
+        ip_address = arg_ip_address
+
+    else:
+
+        # use default IP address for machines in VNET.
+
+        ip_address = '169.254.169.254'
+
+
+
+    # Check if the IP address is valid. If not, try getting the IP address from registry
+
+    # or environment. Exits if no IP address is valid.
+
+    address = make_address(ip_address)
+
+    if not check_ip_address(address, headers):
+
+        print("The provided IP address is invalid or VM is not in VNET. " +
+
+              "Trying registry and environment.")
+
+        ip_address = get_ip_address_reg_env(use_registry)
 
-from util import get_address, get_scheduled_events, post_scheduled_events
+        address = make_address(ip_address)
+
+        if not check_ip_address(address, headers):
+
+            print("Could not find a valid IP address. Please create your VM within a VNET " +
+
+                  "or run discovery.py first")
+
+            exit(1)
+
+    return address
+
+def get_scheduled_events(address, headers):
+
+    '''
+
+    Make a GET request and return the response received.
+
+    '''
+
+    request = urllib.request.Request(address, headers=headers)
+
+    return urllib.request.urlopen(request)
+
+
+
+def post_scheduled_events(address, data, headers):
+
+    '''
+
+    Make a POST request.
+
```

Comment:
Is all this whitespace expected?

Reply:
I just copy-pasted those functions as-is from util.py to sample.py. So, yes, the spaces were part of the original version :-)